components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java

```diff
@@ -81,8 +81,10 @@ public Response loginAppNamePost(String appName
                 refToken = AuthUtil
                         .extractTokenFromHeaders(request.getHeaders(), AuthRESTAPIConstants.WSO2_SP_REFRESH_TOKEN);
                 if (refToken == null) {
-                    LOG.error("Unable to extract refresh token from the header for the request '"
-                            + removeCRLFCharacters(appName));
+                    if (LOG.isDebugEnabled()) {
```

Comment:
No need for check `isDebugEnabled()` with SLF4J

Reply:
Please use "parameterized" log messages. Refer example [here](https://github.com/this/carbon-ui-server/blob/ebae0947615eabb91bfa5fdf3aa8cc6e78fc21ad/components/org.wso2.carbon.uiserver/src/main/java/org/wso2/carbon/uiserver/internal/StartupListener.java#L89)

Reply:
I think we have to else we’ll do unnecessary string concatunations

Reply:
If you use parameterized messages, then any string concatenations won't happen.
`logger.debug("Student '{}' is excellent.", student.getName());`
more info: https://www.slf4j.org/faq.html#logging_performance

Reply:
Fixed in 96ca4ef and 2882642